tests/spec/jasmine-event-recur.js

```diff
@@ -226,6 +233,27 @@ describe("The Calendar Interval", function() {
         expect(valentines.matches( moment(startDate) )).toBe(false);
     });
 
+    it("weeksOfMonthByDay should work combined with daysOfWeek", function() {
+        var recurrence = moment.recur().every(["Sunday"]).daysOfWeek().every([0, 2]).weeksOfMonthByDay();
+        expect(recurrence.matches(moment(startDate))).toBe(false);
+        expect(recurrence.matches(moment(startDate).date(6))).toBe(true);
+        expect(recurrence.matches(moment(startDate).date(8))).toBe(false);
+        expect(recurrence.matches(moment(startDate).date(13))).toBe(false);
+        expect(recurrence.matches(moment(startDate).date(20))).toBe(true);
+        expect(recurrence.matches(moment(startDate).date(27))).toBe(false);
+
+        recurrence = moment.recur().every(["Sunday", "Thursday"]).daysOfWeek().every([1, 3, 4]).weeksOfMonthByDay();
```

Comment:
Also... your weeks are zero indexed... what month has 5 weeks in it :trollface: ?